src/moepkg/highlight.nim

```diff
@@ -0,0 +1,45 @@
+import packages/docutils/highlite
+import strutils, unicode, sequtils
+import ui
+
+proc getHighlightColor(buffer, language: string): seq[seq[ColorPair]] =
+  let lang = getSourceLanguage(language)
+  var token = GeneralTokenizer()
+  token.initGeneralTokenizer(buffer)
+  var color: seq[ColorPair] = @[]
+  let defaultColor = brightWhiteDefault
+  
+  while true:
+    token.getNextToken(lang)
+    let str = buffer[token.start ..< token.start + token.length]
+
+    case token.kind:
+    of gtKeyword:
+      color = concat(color, brightGreenDefault.repeat(str.len))
+    of gtEof:
+      break
+    of gtWhitespace:
+      var numOfSpace = 0
+      for i in 0 ..< str.len:
+        if str[i] != '\n':
+          numOfSpace.inc
+      color = concat(color, defaultColor.repeat(numOfSpace))
+    else:
+      color = concat(color, defaultColor.repeat(str.len))
+
+  var splitBuffer: seq[string] = @[]
+  for buffer in buffer.splitLines:
+    splitBuffer.add(buffer)
+
+  result = @[]
+  var all = 0
+  for i in 0 ..< splitBuffer.len:
+    var line: seq[Colorpair] = @[]
+    for j in 0 ..< splitBuffer[i].len:
+      if splitBuffer[i].len != 0:
```

Comment:
When throughing this line, splitBuffer[i].len isn't equal to 0.
So this condition is meaningless .